documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -11,203 +11,683 @@ Proposal API for backend-agnostic, refactored Input via keyboards, mice, and con
+    Button7,
+    Button8,
+    Button9,
+    Button10,
+    Button11,
+    Button12,
+    Button13,
+    Button14,
+    Button15,
+    Button16,
+    Button17,
+    Button18,
+    Button19,
+    Button20,
+    Button21,
+    Button22,
+    Button23,
+    Button24,
+    Button25,
+    Button26,
+    Button27,
+    Button28,
+    Button29,
+    Button30,
+    Button31
+}
+```
+
+## Cursor Configuration
+
+`ICursorConfiguration` is defined as:
+
+```cs
+public interface ICursorConfiguration
+{
+    CursorModes SupportedModes { get; }
+    CursorModes Mode { get; set; }
+    CursorStyles SupportedStyles { get; }
+    CursorStyles Style { get; set; }
+    CursorFlags SupportedFlags { get; }
+    CursorFlags Flags { get; set; }
+    RawImage? Image { get; set; }
+}
+```
+
+Please note that the `Hotspot` properties present in 1.X and 2.0 have been removed given that they didn't do anything, only providing a place for the developer to store that information and use it if they want to. If the developer wants this, they should use their own variables instead.
+
+`SupportedModes` is a bitmask containing all of the cursor modes that are supported by this backend. This must be queried before setting `Mode` - the currently active cursor mode. An exception should be thrown if an attempt is made to set `Mode` to an unsupported mode or multiple modes (i.e. multiple bits set).
+
+`SupportedStyles` is a bitmask containing all of the cursor styles that are supported by this backend. This must be queried before setting `Style` - the currently active cursor style. An exception should be thrown if an attempt is made to set `Style` to an unsupported style or multiple styles (i.e. multiple bits set).
+
+`Image` uses `RawImage` as-is from Silk.NET.Core, and when set to a non-null value implicitly sets `Style` to custom. As such, you must query `SupportedStyles` before using this property as well. Setting `Image` to `null` will set `Style` back to a standard cursor style, defined by the implementation. It is therefore recommended you set `Style` explicitly when disabling a custom cursor. Note that setting `Style` to a non-`Custom` value will also implicitly set this property to `null`. Setting `Mode` **to** `Custom` explicitly is undefined behaviour, as `Image` won't be set at the time of setting `Mode`.
+
+`SupportedFlags` is a bitmask containing other supported options for the cursor which can be mixed and matched if supported. This must be queried before setting `Flags` - the currently active options. An exception should be thrown if an attempt is made to set an unsupported flag on `Flags`. Unlike the other properties, `Flags` can have multiple bits set.
+
+`Flags` replaces `IsConfined`, which was undefined behaviour on platforms where it was not supported in 2.X - now with `SupportedFlags` you can query its support before setting it.
+
+### Enums
+
+```cs
+[Flags]
+public enum CursorModes
+{
+    Normal,
+    Hidden = 1 << 0,
+    Disabled = 1 << 1,
+    Raw = 1 << 2
```

Comment:
Are these numbers truly part of the proposal? Feels like they can be removed.
In addition, "Normal" needs a bit as well (for checking & representation in supported modes)

Reply:
These numbers should be required, yes. No bits set being no non-standard modes set (every backend must support `Normal`) and specifying "each value must have only one bit set" feels naff compared to... just setting it here? We're programmers, we can read code. 

If you really feel strongly about this we can just reserve the right to change the values, but I think even that has potential to annoy people using serializers for example.